aleph/search/parser.py

```diff
@@ -132,27 +132,7 @@ def __init__(self, args, authz, limit=None):
         # index.
         self.facet_names = set(self.getlist("facet"))
 
-        # Query to use for highlighting, defaults to the search query
-        self.highlight_text = self.get("highlight_text", self.text)
-        # Include highlighted fragments of matching text in the result.
-        self.highlight = self.getbool("highlight", False)
-        self.highlight = self.highlight and SETTINGS.RESULT_HIGHLIGHT
-        self.highlight = self.highlight and self.highlight_text
-        # Length of each snippet in characters
-        self.highlight_length = self.getint("highlight_length", 120)
-        # Number of snippets per document, 0 = return full document text.
         self.highlight_count = self.getint("highlight_count", 3)
-        # By default, the maximum number of characters analyzed for a highlight
-        # request is bounded by the value defined in the
-        # index.highlight.max_analyzed_offset setting (1000000 by default),
-        # and when the number of characters exceeds this limit an error is
-        # returned. By setting `max_analyzed_offset` to a non-negative value
-        # lower than `index.highlight.max_analyzed_offset`, the highlighting
-        # stops at this defined maximum limit, and the rest of the text is not
-        # processed, thus not highlighted and no error is returned.
-        self.max_highlight_analyzed_offset = self.getint(
-            "max_highlight_analyzed_offset", 999999
-        )
```

Comment:
These settings are still relevant, even when using a different highlighter?

For example `highlight_text` is required to be able to highlight the search query in the plaintext viewer:

* Search within a multi-page document
* Go to one of the pages containing the search query
* Switch to the plaintext view, the search query is highlighted
* Reload the page
* Highlight is not visible anymore

And I guess `max_highlight_analyzed_offset` is necessary to protect against errors for large documents.